src/test/java/org/commcare/formplayer/tests/BaseTestClass.java

```diff
@@ -769,4 +818,20 @@ public int getTimezoneOffsetMillis() {
         }
     }
 
+    /**
+     * Utility method to extract the 'installReference' field from test JSON files
+     * and map the remaining JSON fields to the given bean type.
+     *
+     * @param requestPath classpath relative path to JSON file
+     * @param clazz Bean class map JSON to
+     * @return Pair of 'installReference' and deserialized JSON bean
+     * @throws IOException
+     */
+    private <T> Pair<String, T> getInstallReferenceAndBean(String requestPath, Class<T> clazz) throws IOException {
```

Comment:
The alternative to this is to remove the 'installReference' from the test JSON files and pass it in as a separate parameter. This seemed OK for now but it would probably be better to do that in the long run so that we don't mix request bean data with other data.

Reply:
Agreed, and this honestly isn't the hackiest part of tests, really.